internal/ioutil/io.go

```diff
@@ -44,12 +52,38 @@ func ReadAll(r io.Reader, desc ocispec.Descriptor) ([]byte, error) {
 		return nil, errors.New("digest verification failed")
 	}
 
-	// ensure EOF
+	if err := ensureEOF(r); err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// Copy copies from src to dst until either EOF is reached on src or an error occurs.
+// The copied content is verified against the size and the digest.
+func Copy(dst io.Writer, src io.Reader, desc ocispec.Descriptor) error {
+	// verify while copying
+	verifier := desc.Digest.Verifier()
+	mw := io.MultiWriter(verifier, dst)
+	buf := bufPool.Get().(*[]byte)
+	if _, err := io.CopyBuffer(mw, src, *buf); err != nil {
+		return errors.New("copy failed")
+	}
+	bufPool.Put(buf)
```

Comment:
It doesn't really matter. I usually add "defer bufPool.Put(buf)" right after `bufPool.Get()`.